runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,585 @@
+		if err != nil {
+			// TODO (paralta) If image not required this should not be fatal
+			errs <- provider.FatalError{
+				Err: err,
+			}
+			return
+		}
+
+		asset := models.AssetType{}
+		err = asset.FromContainerInfo(models.ContainerInfo{
+			ContainerName: utils.PointerTo(inspect.Name),
+			CreatedAt:     utils.PointerTo(created),
+			Id:            utils.PointerTo(container.ID),
+			Image:         utils.PointerTo(imageInfo),
+			Labels:        convertTags(container.Labels),
+			Location:      nil, // TODO (paralta) Clarify what is location
+			ObjectType:    "ContainerInfo",
+		})
+		if err != nil {
+			errs <- provider.FatalError{
+				Err: fmt.Errorf("failed to create AssetType from ContainerInfo. Provider=%s: %w", models.Docker, err),
+			}
+			return
+		}
+		assets <- asset
+	}
+}
+
+func (c Client) RunAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+
+	scanName, err := getScanName(config)
+	if err != nil {
+		return provider.FatalError{Err: err}
+	}
+	resp, err := c.dockerClient.VolumeList(ctx, volume.ListOptions{
+		Filters: filters.NewArgs(filters.Arg("name", scanName)),
+	})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to get volumes. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	if len(resp.Volumes) == 0 {
+		_, err = c.dockerClient.VolumeCreate(ctx, volume.CreateOptions{
+			Name: scanName,
+		})
+		if err != nil {
+			return provider.FatalError{
+				Err: fmt.Errorf("failed to create volume. Provider=%s: %w", models.Docker, err),
+			}
+		}
+	}
+
+	readCloser, err := c.export(ctx, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to export target. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	// Create an ephemeral container to populate volume with export output
+	_, err = c.dockerClient.ImagePull(ctx, "alpine", types.ImagePullOptions{})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to pull helper image. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	response, err := c.dockerClient.ContainerCreate(ctx,
+		&container.Config{
+			Image: "alpine",
+		},
+		&container.HostConfig{
+			Mounts: []mount.Mount{
+				{
+					Type:   mount.TypeVolume,
+					Source: scanName,
+					Target: "/data",
+				},
+			},
+		}, nil, nil, "")
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to create helper container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	defer func() {
+		err = c.dockerClient.ContainerRemove(ctx, response.ID, types.ContainerRemoveOptions{})
+		if err != nil {
+			_ = fmt.Errorf("failed to remove helper container. Provider=%s: %w", models.Docker, err)
+		}
+	}()
+	err = c.dockerClient.CopyToContainer(ctx, response.ID, "/data", readCloser, types.CopyToContainerOptions{})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to copy data to container. Provider=%s: %w", models.Docker, err),
+		}
+	}
```

Comment:
I think this can be moved to a separate method as well, e.g. `prepareVolume` or something more descriptive?

Reply:
agreed, will change this 😄